### PR TITLE
Revisit the validation

### PR DESCRIPTION
**Pull request by minseok-ryu**
1. debug: validation for BCELoss
2. Validation at a client-level
3. Separate run_serial.py

### CODE REVIEW

**Comment by minseok-ryu** [comment]
Create ``run_serial.py``

File: src/appfl/run_mpi.py [hunk 1]
```diff
@@ -243,7 +123,7 @@ def run_server(
         accuracy = 0
         BestAccuracy = 0
         if cfg.validation == True:
-            test_loss, accuracy = validation(server, server_dataloader)
+            test_loss, accuracy = validation(server, copy.deepcopy(server.model), test_dataloader)
```
Comment (kibaekkim) [inline src/appfl/run_mpi.py:126]: This looks like copying the model every epoch, which may be expensive. Can we avoid this?

File: src/appfl/misc/utils.py [hunk 2]
```diff
@@ -26,26 +29,31 @@ def validation(self, dataloader):
...
+
             correct += pred.eq(target.view_as(pred)).sum().item()
 
     # FIXME: do we need to sent the model to cpu again?
     # self.model.to("cpu")
 
-    test_loss = test_loss / tmpcnt
-    accuracy = 100.0 * correct / tmptotal
+    test_loss = round(test_loss / tmpcnt, 4)
+    test_accuracy = round(100.0 * correct / tmptotal, 2)
```
Comment (kibaekkim) [inline src/appfl/misc/utils.py:46]: Why do we have to round the numbers?

**Review by kibaekkim — COMMENTED** [review]
I left some comments. 

For another minor comment, I think in general (maybe by some naming conventions) variable names are all lower cases, whereas the names for functions and classes start with upper cases. I would suggest to follow the naming conventions for communities. For example, in the codes I see `PerIter_start` could be `per_iter_start`.

File: src/appfl/run_mpi.py (hunk 1 above)
Comment (minseok-ryu) [reply src/appfl/run_mpi.py:126]: Yes. We can insert "model.train()" for every iteration.

File: src/appfl/misc/utils.py (hunk 2 above)
Comment (minseok-ryu) [reply src/appfl/misc/utils.py]: removed.

File: src/appfl/run_mpi.py (hunk 1 above)
Comment (minseok-ryu) [reply src/appfl/run_mpi.py]: removed.

**Review by kibaekkim — APPROVED** [review]

**Pull request merged**